terraform/evaluate.go

```diff
@@ -758,11 +758,12 @@ func (d *evaluationStateData) GetResource(addr addrs.Resource, rng tfdiags.Sourc
 			}
 
 			// If our provider schema contains sensitive values, mark those as sensitive
+			afterMarks := change.AfterValMarks
 			if schema.ContainsSensitive() {
-				val = markProviderSensitiveAttributes(schema, val)
+				afterMarks = append(afterMarks, getValMarks(schema, val, nil)...)
```

Comment:
There's 1 remaining call to `markProviderSensitiveAttributes` a little further down. Do we need to do the same thing there too?

Reply:
Oh, nm, I missed that `val` was being re-assigned there